### ISSUE TITLE
Issue when trying to set up development environment

### ISSUE THREAD

**Issue by SarvaniVarma** [Apicurio/apicurio-studio#352]
I am trying to set a dev environment for back-end , I got 404 error.
I have followed the process given in the below link,
https://apicurio-studio.readme.io/v0.1.8/docs/setting-up-a-development-environment
I got 404 error when accessing the application on https://localhost:8443 and http://localhost:8080.
Maven build was successful, but I get 404 Not Found error.
Could you please help.
![screenshot](https://user-images.githubusercontent.com/41433069/42941606-e64cb586-8b7a-11e8-8b79-e755c9e042c4.png)

**Comment by SarvaniVarma** [comment]
Hi Eric,
Console output from wildfly server:

18:21:01,584 INFO  [org.jboss.modules] (main) JBoss Modules version 1.5.2.Final
18:21:01,771 INFO  [org.jboss.msc] (main) JBoss MSC version 1.2.6.Final
18:21:01,840 INFO  [org.jboss.as] (MSC service thread 1-6) WFLYSRV0049: WildFly Full 10.1.0.Final (WildFly Core 2.2.0.Final) starting
18:21:01,842 DEBUG [org.jboss.as.config] (MSC service thread 1-6) Configured system properties:
	awt.toolkit = sun.awt.X11.XToolkit
	file.encoding = UTF-8
	file.encoding.pkg = sun.io
	file.separator = /
	java.awt.graphicsenv = sun.awt.X11GraphicsEnvironment
	java.awt.headless = true
	java.awt.printerjob = sun.print.PSPrinterJob
	java.class.path = /home/tcs/Downloads/apicurio-studio-0.2.12.Final/jboss-modules.jar:/usr/lib/jvm/java-8-oracle/jre/lib/resources.jar:/usr/lib/jvm/java-8-oracle/jre/lib/rt.jar:/usr/lib/jvm/java-8-oracle/jre/lib/jsse.jar:/usr/lib/jvm/java-8-oracle/jre/lib/jce.jar:/usr/lib/jvm/java-8-oracle/jre/lib/charsets.jar:/usr/lib/jvm/java-8-oracle/jre/lib/jfr.jar
	java.class.version = 52.0
	java.endorsed.dirs = /usr/lib/jvm/java-8-oracle/jre/lib/endorsed
	java.ext.dirs = /usr/lib/jvm/java-8-oracle/jre/lib/ext:/usr/java/packages/lib/ext
	java.home = /usr/lib/jvm/java-8-oracle/jre
	java.io.tmpdir = /tmp
	java.library.path = /usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
	java.net.preferIPv4Stack = true
	java.runtime.name = Java(TM) SE Runtime Environment
	java.runtime.version = 1.8.0_151-b12
	java.specification.name = Java Platform API Specification
	java.specification.vendor = Oracle Corporation
	java.specification.version = 1.8
	java.util.logging.manager = org.jboss.logmanager.LogManager
	java.vendor = Oracle Corporation
	java.vendor.url = http://java.oracle.com/
	java.vendor.url.bug = http://bugreport.sun.com/bugreport/
	java.version = 1.8.0_151
	java.vm.info = mixed mode
	java.vm.name = Java HotSpot(TM) 64-Bit Server VM
	java.vm.specification.name = Java Virtual Machine Specification
	java.vm.specification.vendor = Oracle Corporation
	java.vm.specification.version = 1.8
	java.vm.vendor = Oracle Corporation
	java.vm.version = 25.151-b12
	javax.management.builder.initial = org.jboss.as.jmx.PluggableMBeanServerBuilder
	javax.xml.datatype.DatatypeFactory = __redirected.__DatatypeFactory
	javax.xml.parsers.DocumentBuilderFactory = __redirected.__DocumentBuilderFactory
	javax.xml.parsers.SAXParserFactory = __redirected.__SAXParserFactory
	javax.xml.stream.XMLEventFactory = __redirected.__XMLEventFactory
	javax.xml.stream.XMLInputFactory = __redirected.__XMLInputFactory
	javax.xml.stream.XMLOutputFactory = __redirected.__XMLOutputFactory
	javax.xml.transform.TransformerFactory = __redirected.__TransformerFactory
	javax.xml.validation.SchemaFactory:http://www.w3.org/2001/XMLSchema = __redirected.__SchemaFactory
	javax.xml.xpath.XPathFactory:http://java.sun.com/jaxp/xpath/dom = __redirected.__XPathFactory
	jboss.bind.address = 10.125.176.50
	jboss.bind.address.management = 10.125.176.50
	jboss.home.dir = /home/tcs/Downloads/apicurio-studio-0.2.12.Final
	jboss.host.name = tcs
	jboss.modules.dir = /home/tcs/Downloads/apicurio-studio-0.2.12.Final/modules
	jboss.modules.system.pkgs = org.jboss.byteman
	jboss.node.name = tcs
	jboss.qualified.host.name = tcs
	jboss.server.base.dir = /home/tcs/Downloads/apicurio-studio-0.2.12.Final/standalone
	jboss.server.config.dir = /home/tcs/Downloads/apicurio-studio-0.2.12.Final/standalone/configuration
	jboss.server.data.dir = /home/tcs/Downloads/apicurio-studio-0.2.12.Final/standalone/data
	jboss.server.deploy.dir = /home/tcs/Downloads/apicurio-studio-0.2.12.Final/standalone/data/content
	jboss.server.log.dir = /home/tcs/Downloads/apicurio-studio-0.2.12.Final/standalone/log
	jboss.server.name = tcs
	jboss.server.persist.config = true
	jboss.server.temp.dir = /home/tcs/Downloads/apicurio-studio-0.2.12.Final/standalone/tmp
	line.separator = 

	logging.configuration = file:/home/tcs/Downloads/apicurio-studio-0.2.12.Final/standalone/configuration/logging.properties
	module.path = /home/tcs/Downloads/apicurio-studio-0.2.12.Final/modules
	org.jboss.boot.log.file = /home/tcs/Downloads/apicurio-studio-0.2.12.Final/standalone/log/boot.log
	org.jboss.logmanager.nocolor = true
	org.jboss.resolver.warning = true
	org.xml.sax.driver = __redirected.__XMLReaderFactory
	os.arch = amd64
	os.name = Linux
	os.version = 4.15.0-24-generic
	path.separator = :
	program.name = JBossTools: WildFly 10.x
	sun.arch.data.model = 64
	sun.boot.class.path = /usr/lib/jvm/java-8-oracle/jre/lib/resources.jar:/usr/lib/jvm/java-8-oracle/jre/lib/rt.jar:/usr/lib/jvm/java-8-oracle/jre/lib/sunrsasign.jar:/usr/lib/jvm/java-8-oracle/jre/lib/jsse.jar:/usr/lib/jvm/java-8-oracle/jre/lib/jce.jar:/usr/lib/jvm/java-8-oracle/jre/lib/charsets.jar:/usr/lib/jvm/java-8-oracle/jre/lib/jfr.jar:/usr/lib/jvm/java-8-oracle/jre/classes
	sun.boot.library.path = /usr/lib/jvm/java-8-oracle/jre/lib/amd64
	sun.cpu.endian = little
	sun.cpu.isalist = 
	sun.desktop = gnome
	sun.io.unicode.encoding = UnicodeLittle
	sun.java.command = org.jboss.modules.Main -mp /home/tcs/Downloads/apicurio-studio-0.2.12.Final/modules org.jboss.as.standalone -b 10.125.176.50 --server-config=standalone-apicurio.xml -Djboss.server.base.dir=/home/tcs/Downloads/apicurio-studio-0.2.12.Final/standalone
	sun.java.launcher = SUN_STANDARD
	sun.jnu.encoding = UTF-8
	sun.management.compiler = HotSpot 64-Bit Tiered Compilers
	sun.os.patch.level = unknown
	sun.rmi.dgc.client.gcInterval = 3600000
	sun.rmi.dgc.server.gcInterval = 3600000
	user.country = IN
	user.dir = /home/tcs/Downloads/apicurio-studio-0.2.12.Final/bin
	user.home = /home/tcs
	user.language = en
	user.name = tcs
	user.timezone = Asia/Kolkata
18:21:01,842 DEBUG [org.jboss.as.config] (MSC service thread 1-6) VM Arguments: -Dprogram.name=JBossTools: WildFly 10.x -Xms64m -Xmx512m -Dorg.jboss.resolver.warning=true -Djava.net.preferIPv4Stack=true -Dsun.rmi.dgc.client.gcInterval=3600000 -Dsun.rmi.dgc.server.gcInterval=3600000 -Djboss.modules.system.pkgs=org.jboss.byteman -Djava.awt.headless=true -Dorg.jboss.boot.log.file=/home/tcs/Downloads/apicurio-studio-0.2.12.Final/standalone/log/boot.log -Dlogging.configuration=file:/home/tcs/Downloads/apicurio-studio-0.2.12.Final/standalone/configuration/logging.properties -Djboss.home.dir=/home/tcs/Downloads/apicurio-studio-0.2.12.Final -Dorg.jboss.logmanager.nocolor=true -Djboss.bind.address.management=10.125.176.50 -Dfile.encoding=UTF-8 
18:21:02,904 INFO  [org.jboss.as.server] (Controller Boot Thread) WFLYSRV0039: Creating http management service using socket-binding (management-http)
18:21:02,926 INFO  [org.xnio] (MSC service thread 1-5) XNIO version 3.4.0.Final
18:21:02,960 INFO  [org.xnio.nio] (MSC service thread 1-5) XNIO NIO Implementation Version 3.4.0.Final
18:21:03,003 INFO  [org.jboss.remoting] (MSC service thread 1-5) JBoss Remoting version 4.0.21.Final
18:21:03,036 INFO  [org.wildfly.extension.io] (ServerService Thread Pool -- 38) WFLYIO001: Worker 'default' has auto-configured to 8 core threads with 64 task threads based on your 4 available processors
18:21:03,040 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 39) WFLYCLINF0001: Activating Infinispan subsystem.
18:21:03,051 INFO  [org.jboss.as.connector.subsystems.datasources] (ServerService Thread Pool -- 34) WFLYJCA0004: Deploying JDBC-compliant driver class org.h2.Driver (version 1.3)
18:21:03,106 INFO  [org.jboss.as.connector] (MSC service thread 1-5) WFLYJCA0009: Starting JCA Subsystem (WildFly/IronJacamar 1.3.4.Final)
18:21:03,116 INFO  [org.jboss.as.naming] (ServerService Thread Pool -- 47) WFLYNAM0001: Activating Naming Subsystem
18:21:03,117 WARN  [org.jboss.as.txn] (ServerService Thread Pool -- 55) WFLYTX0013: Node identifier property is set to the default value. Please make sure it is unique.
18:21:03,127 INFO  [org.jboss.as.connector.deployers.jdbc] (MSC service thread 1-3) WFLYJCA0018: Started Driver service with driver-name = h2
18:21:03,130 INFO  [org.jboss.as.security] (ServerService Thread Pool -- 54) WFLYSEC0002: Activating Security Subsystem
18:21:03,209 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-3) WFLYUT0003: Undertow 1.4.0.Final starting
18:21:03,219 INFO  [org.jboss.as.jsf] (ServerService Thread Pool -- 45) WFLYJSF0007: Activated the following JSF Implementations: [main]
18:21:03,227 INFO  [org.jboss.as.webservices] (ServerService Thread Pool -- 57) WFLYWS0002: Activating WebServices Extension
18:21:03,250 INFO  [org.jboss.as.security] (MSC service thread 1-6) WFLYSEC0001: Current PicketBox version=4.9.6.Final
18:21:03,327 INFO  [org.jboss.as.naming] (MSC service thread 1-2) WFLYNAM0003: Starting Naming Service
18:21:03,327 INFO  [org.jboss.as.mail.extension] (MSC service thread 1-6) WFLYMAIL0001: Bound mail session [java:jboss/mail/Default]
18:21:03,376 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 56) WFLYUT0014: Creating file handler for path '/home/tcs/Downloads/apicurio-studio-0.2.12.Final/welcome-content' with options [directory-listing: 'false', follow-symlink: 'false', case-sensitive: 'true', safe-symlink-paths: '[]']
18:21:03,379 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-5) WFLYUT0012: Started server default-server.
18:21:03,380 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-5) WFLYUT0018: Host default-host starting
18:21:03,511 INFO  [org.jboss.as.ejb3] (MSC service thread 1-6) WFLYEJB0481: Strict pool slsb-strict-max-pool is using a max instance size of 64 (per class), which is derived from thread worker pool sizing.
18:21:03,511 INFO  [org.jboss.as.ejb3] (MSC service thread 1-2) WFLYEJB0482: Strict pool mdb-strict-max-pool is using a max instance size of 16 (per class), which is derived from the number of CPUs on this host.
18:21:03,533 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-5) WFLYUT0006: Undertow HTTP listener default listening on 10.125.176.50:8080
18:21:03,664 INFO  [org.jboss.as.connector.subsystems.datasources] (MSC service thread 1-7) WFLYJCA0001: Bound data source [java:jboss/datasources/ExampleDS]
18:21:03,680 INFO  [org.jboss.as.connector.subsystems.datasources] (MSC service thread 1-8) WFLYJCA0001: Bound data source [java:jboss/datasources/ApicurioDS]
18:21:03,863 INFO  [org.jboss.as.server.deployment.scanner] (MSC service thread 1-2) WFLYDS0013: Started FileSystemDeploymentService for directory /home/tcs/Downloads/apicurio-studio-0.2.12.Final/standalone/deployments
18:21:04,009 INFO  [org.infinispan.factories.GlobalComponentRegistry] (MSC service thread 1-8) ISPN000128: Infinispan version: Infinispan 'Chakra' 8.2.4.Final
18:21:04,062 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 67) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
18:21:04,065 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 67) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
18:21:04,060 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 63) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
18:21:04,067 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 63) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
18:21:04,069 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 65) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
18:21:04,070 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 65) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
18:21:04,204 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-5) WFLYUT0006: Undertow HTTPS listener https listening on 10.125.176.50:8443
18:21:04,259 INFO  [org.jboss.ws.common.management] (MSC service thread 1-7) JBWS022052: Starting JBossWS 5.1.5.Final (Apache CXF 3.1.6) 
18:21:04,377 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0060: Http management interface listening on http://10.125.176.50:9990/management
18:21:04,377 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0051: Admin console listening on http://10.125.176.50:9990
18:21:04,377 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0025: WildFly Full 10.1.0.Final (WildFly Core 2.2.0.Final) started in 3184ms - Started 338 of 584 services (394 services are lazy, passive or on-demand)



Modules added to my server 

![modules](https://user-images.githubusercontent.com/41433069/42943911-d1a6e302-8b81-11e8-88b1-b413780657de.png)

**Comment by EricWittmann** [comment]
OK so you should only have the "apicurio-studio-platforms-**wildfly**-*" modules added.  The swarm modules shoudl **not** be included.

The output indicates that none of the Apicurio modules were deployed to Wildfly.  Check the **standalone/deployments** directory in Wildfly to see if anything has been deployed there.  The documentation instructions say to clean that directory out - the reason is that Eclipse should publish the modules into that directory for you.  So for example, mine looks like this:

![image](https://user-images.githubusercontent.com/1890703/42945985-d2c80030-8b37-11e8-9591-9484040f5e73.png)

Those folders were all put there by Eclipse.  If they are not present, then it's not working. :)

**Comment by SarvaniVarma** [comment]
Hi Eric,
I have set up dev environment for both front-end and back-end, and also I have made few changes in front-end, the changes got reflected in the front-end set up but when I tried to run the same front-end changes in the back-end dev environment, the changes are not getting reflected. I am new to Apicurio, Some help will be appreciated.

**Comment by EricWittmann** [comment]
Yes that's expected behavior - if you want your front end changes to show up when you access the application running in Wildfly (the Quickstart) then you need to do a command line build in maven.  Or alternatively we would need to come up with a new (faster) procedure where we do an angular production build and copy the resulting `dist` directory into Wildfly.

Are you simply trying to verify that your changes will work when built and deployed to Wildfly, or is there another use-case you need this for?

**Comment by EricWittmann** [comment]
OK actually it occurred to me that I had a script that would do exactly this.  So I've updated the script to make it more generic (it was originally just a local script I used for my own system) and have checked it in:

https://github.com/Apicurio/apicurio-studio/blob/master/front-end/studio/deploy-prod.sh

Run that script from its location and it should build the front-end application and copy it to Wildfly in the appropriate location.  I don't think you even need to restart Wildfly for the changes to work - just reload your browser.

Let me know how it goes! :)
